### PR TITLE
Update Go dependencies

### PR DESCRIPTION
**Pull request by ph**
## What is the problem this PR solves?

Some of the `indirect` dependencies are required to be updated.

- Update beats to latest master and the indirect dependencies
- Move from jwt-go golang-jwt/jwt (new maintained version)
- add overrides for goe
- add CC0 to list of accepted license.

## Checklist

~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
-[x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Comment by ph** [comment]
This will require more works, see https://github.com/elastic/beats/pull/28716 for some of the pain.

File: go.mod [hunk 1]
```diff
@@ -5,13 +5,13 @@ go 1.17
 require (
 	github.com/Pallinder/go-randomdata v1.2.0
 	github.com/dgraph-io/ristretto v0.1.0
-	github.com/elastic/beats/v7 v7.0.0-alpha2.0.20211207150041-4641f759226c
+	github.com/elastic/beats/v7 v7.17.1
```
Comment (jlind23) [inline go.mod:8]: Why do we rely on 7.17.1 as there is already a 7.17.2?

Comment (ph) [reply go.mod:8]: My local copy is more recent, let me push the last changes.

Comment (ph) [reply go.mod]: This is blocked by https://github.com/elastic/elastic-agent/pull/273, fleet server has dependencies on some of the packages from Elastic Agent. They used to be in the main beats repository.

```
 ❯ go mod tidy                                                                                                                                                          [15:17:22]
go: finding module for package github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/server
go: finding module for package github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/logger
go: finding module for package github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/packer
```

Once everything is updated, I will make an issue to reduce the dependencies between the three repositories.

**Review by jlind23 — APPROVED** [review]

**Pull request merged**